ckan/controllers/group.py

```diff
@@ -133,6 +131,21 @@ def _guess_group_type(self, expecting_name=False):
 
         return gt
 
+    def _ensure_controller_matches_group_type(self, id):
+        group = model.Group.get(id)
+        if group is None:
+            abort(404, _('Group not found'))
+        if group.type not in self.group_types:
+            abort(404, _('Incorrect group type'))
+        return group.type
```

Comment:
This method is called at the start of most controller methods, to stop it working if the user tries using /organization/mygroup instead of /group/mygroup etc.